alf/algorithms/sac_algorithm.py

```diff
@@ -773,7 +847,8 @@ def _critic_train_step(self, inputs: TimeStep, state: SacCriticState,
             target_critics = torch.sum(
                 discrete_act_dist.probs * target_critics, dim=-1)
 
-        target_critic = target_critics.reshape(inputs.reward.shape)
+        target_critic = target_critics.reshape(target_critics.shape[0],
+                                               *self._reward_spec.shape)
```

Comment:
Why is this change necessary?

Reply:
This is because inputs.reward is no longer available